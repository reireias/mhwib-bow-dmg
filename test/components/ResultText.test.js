import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import target from '@/components/ResultText'
import { createStore, buildResult } from '@/test/helper'

Vue.use(Vuetify)
const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

describe('instance', () => {
  let store
  beforeEach(() => {
    store = createStore()
  })

  it('should be vue instance', () => {
    const vuetify = new Vuetify()
    const wrapper = shallowMount(target, {
      localVue,
      store,
      vuetify,
      propsData: { result: buildResult() },
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
