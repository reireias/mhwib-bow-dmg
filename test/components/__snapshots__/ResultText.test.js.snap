// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instance should be vue instance 1`] = `
"<div>
  <v-textarea-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" value=\\"武器：銀火竜の烈弓
カスタム：なし
バフ：護符、爪、粉塵、食事大、種
スキル：弱点特効3、属性強化6
モーション：溜め1 [7×1, 属性補正0.2]
ターゲット：柱/荷車 全身 [弾80, 火30]

結果
通常時：物理: 10, 属性: 10
会心時：物理: 20, 属性: 20
期待値：物理: 15, 属性: 15
合計：75\\" backgroundcolor=\\"\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" type=\\"text\\" autogrow=\\"true\\" rowheight=\\"24\\" rows=\\"5\\"></v-textarea-stub>
  <center>
    <v-btn-stub tag=\\"button\\" activeclass=\\"\\" type=\\"button\\">コピー</v-btn-stub>
  </center>
  <v-snackbar-stub timeout=\\"6000\\">コピーしました</v-snackbar-stub>
</div>"
`;
